src/System.Private.CoreLib/shared/System/Diagnostics/Tracing/DiagnosticCounter.cs

```diff
@@ -75,9 +76,29 @@ public void AddMetadata(string key, string value)
             }
         }
 
-        public string? DisplayName { get; set; }
+        private string _displayName = "";
+        public string DisplayName
+        {
+            set
+            {
+                if (value == null)
+                    throw new ArgumentException("Cannot set null as DisplayName");
```

Comment:
Should we update the tests in corefx as well to reflect this behavior change?

Reply:
At the moment there isn't any test that hits this particular case -- I'll be adding one soon